remote/remote.go

```diff
@@ -833,15 +833,19 @@ func runWpCliCmdRemote(conn net.Conn, GUID string, rows uint16, cols uint16, wpC
 				break
 			}
 		}
-		log.Println("closing logfile and marking the WP-CLI as finished")
+		log.Println("closing logfile")
 		logFile.Sync()
 		logFile.Close()
 
 		time.Sleep(time.Duration(50 * time.Millisecond.Nanoseconds()))
-
-		wpcli.padlock.Lock()
-		wpcli.Running = false
-		wpcli.padlock.Unlock()
+		if wpcli.Running {
```

Comment:
I guess we can `.Lock()` before we read `wpcli.Running`, to avoid races there too

Reply:
@alexcriss Sadly it's the `wpcli.padlock.Lock()` that's causing the panic, because it has been nulled outside the goroutine.

Another option would be to check for null, but I preferred using the running part because it made the behavior more clear